extensions/schema-registry/apicurio/avro/deployment/src/main/java/io/quarkus/apicurio/registry/avro/ApicurioRegistryAvroProcessor.java

```diff
@@ -14,27 +15,36 @@ FeatureBuildItem feature() {
+    public void apicurioRegistryAvro(CurateOutcomeBuildItem curateOutcomeBuildItem,
+            BuildProducer<ReflectiveClassBuildItem> reflectiveClass,
             BuildProducer<ExtensionSslNativeSupportBuildItem> sslNativeSupport) {
 
-        reflectiveClass.produce(new ReflectiveClassBuildItem(true, true, false,
-                "io.apicurio.registry.serde.avro.AvroKafkaDeserializer",
-                "io.apicurio.registry.serde.avro.AvroKafkaSerializer"));
-
-        reflectiveClass.produce(new ReflectiveClassBuildItem(true, true, true,
-                "io.apicurio.registry.serde.strategy.SimpleTopicIdStrategy",
-                "io.apicurio.registry.serde.strategy.TopicIdStrategy",
-                "io.apicurio.registry.serde.avro.DefaultAvroDatumProvider",
-                "io.apicurio.registry.serde.avro.ReflectAvroDatumProvider",
-                "io.apicurio.registry.serde.avro.strategy.RecordIdStrategy",
-                "io.apicurio.registry.serde.avro.strategy.TopicRecordIdStrategy"));
-
-        reflectiveClass.produce(new ReflectiveClassBuildItem(true, true, true,
-                "io.apicurio.registry.serde.DefaultSchemaResolver",
-                "io.apicurio.registry.serde.DefaultIdHandler",
-                "io.apicurio.registry.serde.Legacy4ByteIdHandler",
-                "io.apicurio.registry.serde.fallback.DefaultFallbackArtifactProvider",
-                "io.apicurio.registry.serde.headers.DefaultHeadersHandler"));
+        if (curateOutcomeBuildItem.getApplicationModel().getDependencies().stream().anyMatch(
+                x -> x.getGroupId().equalsIgnoreCase("io.apicurio")
+                        && x.getArtifactId().equalsIgnoreCase("apicurio-registry-serdes-avro-serde"))) {
+            reflectiveClass.produce(new ReflectiveClassBuildItem(true, true, false,
+                    "io.apicurio.registry.serde.avro.AvroKafkaDeserializer",
+                    "io.apicurio.registry.serde.avro.AvroKafkaSerializer"));
+
+            reflectiveClass.produce(new ReflectiveClassBuildItem(true, true, true,
+                    "io.apicurio.registry.serde.avro.DefaultAvroDatumProvider",
+                    "io.apicurio.registry.serde.avro.ReflectAvroDatumProvider",
+                    "io.apicurio.registry.serde.avro.strategy.RecordIdStrategy",
+                    "io.apicurio.registry.serde.avro.strategy.TopicRecordIdStrategy"));
+        }
+
+        if (curateOutcomeBuildItem.getApplicationModel().getDependencies().stream().anyMatch(
+                x -> x.getGroupId().equalsIgnoreCase("io.apicurio")
+                        && x.getArtifactId().equalsIgnoreCase("apicurio-registry-serde-common"))) {
+            reflectiveClass.produce(new ReflectiveClassBuildItem(true, true, true,
+                    "io.apicurio.registry.serde.strategy.SimpleTopicIdStrategy",
+                    "io.apicurio.registry.serde.strategy.TopicIdStrategy",
+                    "io.apicurio.registry.serde.DefaultSchemaResolver",
```

Comment:
Do you confirm that you don't test the version on purpose? You accept to get the warning in case the version is 2.2+, right?

Reply:
No, that slipped away as well. Thanks for the catch!

Reply:
Revisiting this I see that quarkus [is using 2.3.1.Final](https://github.com/quarkusio/quarkus/blob/65c3f7df50da58a1840629ada95f990666e46278/bom/application/pom.xml#L200) but in integration-tests/kafka-avro the version [is pinned to 1.2.2.Final](https://github.com/quarkusio/quarkus/blob/65c3f7df50da58a1840629ada95f990666e46278/integration-tests/kafka-avro/pom.xml#L17)

My understanding is that we care for boh 1.x and 2.x support. Is it OK to assume we only need to support 2.3.1 from 2.x and "any version" from 1.x? cc @Ladicek @phillip-kruger

Reply:
Ping @Ladicek @phillip-kruger. Can you please have a look at this question? ^^

Reply:
Umm, it's been a while since I worked on this.

So `integration-tests/kafka-avro` verifies support for Apicurio Registry 1.x libraries; the tests for Apicurio Registry 2.x libraries is in `integration-tests/kafka-avro-apicurio2`.

We use Apicurio Registry 1.2.2 libraries in `integration-tests/kafka-avro`, because Apicurio Registry 1.3 libraries require some extra support which is a bit hacky (https://github.com/quarkiverse/quarkus-apicurio-registry-client). I believe we want[ed] to drop support for Apicurio Registry 1.x libraries at some point, together with `integration-tests/kafka-avro`. I'm not sure if that time has come.

For `integration-tests/kafka-avro-apicurio2`, I believe we want to use the same version of Apicurio Registry 2.x libraries that we compile against elsewhere.

Also CC @ozangunalp @carlesarnal

Reply:
Ping @ozangunalp @carlesarnal

Do you have any additional feedback on this? If not I will proceed based on the above comment.

Reply:
No additional feedback other than what Ladislav already said. Apicurio 1.x is no longer supported on our end and users are highly encouraged to move to 2.x. Maybe Quarkus 3 is the perfect moment to drop the 1.x support, but in the end, it's up to the Quarkus maintainers (basically @Ladicek and @ozangunalp in this case, I guess) to make that decision.